addons/mass_mailing/wizard/mailing_mailing_test.py

```diff
@@ -48,10 +57,30 @@ def send_mail_test(self):
                 'notification': True,
                 'mailing_id': mailing.id,
                 'attachment_ids': [(4, attachment.id) for attachment in mailing.attachment_ids],
-                'auto_delete': True,
+                'auto_delete': False,  # they are manually deleted after notifying the document
                 'mail_server_id': mailing.mail_server_id.id,
             }
             mail = self.env['mail.mail'].sudo().create(mail_values)
             mails_sudo |= mail
         mails_sudo.send()
+
+        for mail_sudo in mails_sudo:
+            if mail_sudo.state == 'sent':
+                notification_messages.append(
+                    _('Test mailing successfully sent to %s', mail_sudo.email_to))
+            elif mail_sudo.state == 'exception':
+                notification_messages.append(
+                    _('Test mailing could not be sent to %s:<br>%s',
+                        mail_sudo.email_to,
+                        mail_sudo.failure_reason)
+                )
+
+        # manually delete the emails since we passed 'auto_delete: False'
+        mails_sudo.unlink()
+
+        if notification_messages:
+            self.mass_mailing_id._message_log(body='<ul>%s</ul>' % ''.join(
```

Comment:
Small remark: do we _message_log or _message_post to warn responsible ?

Reply:
As the testing is done from the form view, it's very obvious when something appears in the "tchatter".
I think it's a sufficient feedback, no need to escalate this a lot since it's only a for testing.

Reply:
Good point.